sql/core/src/test/resources/sql-tests/inputs/postgreSQL/numeric.sql

```diff
@@ -657,16 +674,18 @@ SELECT AVG(val) FROM num_data;
 
 -- Check for appropriate rounding and overflow
 CREATE TABLE fract_only (id int, val decimal(4,4)) USING parquet;
-INSERT INTO fract_only VALUES (1, '0.0');
-INSERT INTO fract_only VALUES (2, '0.1');
+-- PostgreSQL implicitly casts string literals to data with decimal types, but
+-- Spark does not support that kind of implicit casts.
+INSERT INTO fract_only VALUES (1, cast('0.0' as decimal(4,4)));
+INSERT INTO fract_only VALUES (2, cast('0.1' as decimal(4,4)));
 -- [SPARK-27923] PostgreSQL throws an exception but Spark SQL is NULL
--- INSERT INTO fract_only VALUES (3, '1.0');	-- should fail
-INSERT INTO fract_only VALUES (4, '-0.9999');
-INSERT INTO fract_only VALUES (5, '0.99994');
+-- INSERT INTO fract_only VALUES (3, '1.0' as decimal(4,4));	-- should fail
+INSERT INTO fract_only VALUES (4, cast('-0.9999' as decimal(4,4)));
+INSERT INTO fract_only VALUES (5, cast('0.99994' as decimal(4,4)));
```

Comment:
So, this is converting `CASTING DURING INSERTING` into `CASTING WITH CAST syntax`.

Reply:
Ur... my fault.. I'll update later.

Reply:
Oh, it was just a comment to clarify. I think it's okay for now `AS-IS` since we don't support the original query in any way.

Reply:
But, the original queries throw unsupported exceptions now, so most of queries in this file become invalid in Spark. In the last commit, I changed these string literals into floating-point literals for depending on INSERT assignment casts. Which do you think is better, `as-is` or `the last commit`?